### PR TITLE
fix: ci test failures with torch 2.5.1 compatibility

### PR DESCRIPTION
**Pull request by xlu451**
Fix CI test failures caused by torch 2.5.1.

### CODE REVIEW

File: .github/workflows/linux-cp310.yaml [hunk 1]
```diff
@@ -28,11 +28,11 @@ jobs:
           pip install -U pip setuptools
       - name: Lint
         run: |
+          pip install .[lint]
```
Comment (bddppq) [inline .github/workflows/linux-cp310.yaml:31]: why swapping the installation order?

File: pyproject.toml [hunk 2]
```diff
@@ -47,7 +47,7 @@ test = [
 runtime = [
     "accelerate", # needed by low_cpu_mem_usage
     "huggingface_hub",
-    "diffusers[torch]>=0.27.0",
+    "diffusers>=0.27.0",
```
Comment (bddppq) [inline pyproject.toml:50]: why?

**Review by bddppq — APPROVED** [review]

File: .github/workflows/linux-cp310.yaml (hunk 1 above)
Comment (xlu451) [reply .github/workflows/linux-cp310.yaml:31]: Running `pip install`. after `pip install torch torchvision torchaudio --index-url https://download.pytorch.org/whl/cpu`  will reinstall torch, causing torch version mismatches with torchvision and torchaudio, resulting in compatibility issues.

File: pyproject.toml (hunk 2 above)
Comment (xlu451) [reply pyproject.toml:50]: Considering the necessity of including both "diffusers[torch]>=0.27.0" and "torch" in the dependencies.

Comment (xlu451) [reply pyproject.toml:50]: diffusers[torch]>=0.27.0 will try to install torch==2.4.1

File: .github/workflows/linux-cp310.yaml (hunk 1 above)
Comment (xlu451) [reply .github/workflows/linux-cp310.yaml]: Tested and confirmed this change is unnecessary.

**Pull request merged**